packages/core/core/src/config.ts

```diff
@@ -23,11 +22,14 @@ import {
   getExternalContractUrl
 } from 'embark-utils';
 import { Logger } from 'embark-logger';
+import { readJsonSync } from 'fs-extra';
 const cloneDeep = require('lodash.clonedeep');
 const { replaceZeroAddressShorthand } = AddressUtils;
 
 import { getBlockchainDefaults, getContractDefaults } from './configDefaults';
 
+const constants = readJsonSync(path.join(__dirname, '../constants.json'));
```

Comment:
Good question. It used to work to require/import it. But, with the `tsconfig.json` files in place that we need for the project references and output in `dist` that matches up structurally with babel's output (package by package), it's important for `"rootDir"` to be specified in `tsconfig.json`. What happens then is that `tsc` will strictly not allow any import from a path that's not inside `rootDir`. So my workaround is to simply read the file.

Reply:
@emizzle the reason that's not a good solution is that the `rootDir` would have to be `"./"` instead of `"./src"` and then `tsc` will structure the declarations output in `dist/` like this:

```
dist
├── constants.d.ts
├── src
│   ├── index.d.ts
│   ├── index.d.ts.map
│   ├── ...
```

Two problems with that:

1. It doesn't match up with babel's output of the transpiled TS, which looks like this:

```
dist
├── index.js
├── index.js.map
├── ...
```

We could live with that (and an adjustment in package.json to have `"types": "./dist/src/index.d.ts"`) but I don't like it very much — it seems more natural for `index.js` and `index.d.ts` to be siblings in `dist/`.

2. We'll have a declarations file `dist/constants.d.ts` which seems... weird. `constants.json` is just JSON, it's not actually a module *per se*.

*Important note:* in modules that are external to `embark-core`, it's not a problem to do:
```
import constants from 'embark-core/constants.json';
```

Where this issue arises is when `.ts` sources in the `rootDir` need to load a JSON file in that same package but that file is outside of `rootDir`. There are several places in this PR where it comes up (in various packages), involving `constants.json`, `suggestions.json`, and `package.json`. I think the simplest solution even if it's a little awkward, it to load with `readJsonSync`, taking care to join the path with `__dirname` because unlike `require` and `import` where the path specified is always interpreted as relative to the module, when you pass a relative path to one of the `fs` APIs it's interpreted as relative to `process.cwd()`.